Zilliqa/contracts/compressed/minimize_proxy.scilla

```diff
@@ -35,14 +36,41 @@ in
 
 contract ProxyContract
 (
+contract_owner: ByStr20,
+name :    String,
+symbol :  String,
+decimals : Uint32,
+init_supply : Uint128,
 init_implementation : ByStr20,
 init_admin : ByStr20
 )
+with
+let string_is_not_empty =
+fun (s : String) =>
+let zero = Uint32 0 in
+let s_length = builtin strlen s in
+let s_empty = builtin eq s_length zero in
+negb s_empty
+in
+let name_ok = string_is_not_empty name in
+let symbol_ok = string_is_not_empty symbol in
+let name_symbol_ok = andb name_ok symbol_ok in
+let decimals_ok =
+let six = Uint32 6 in
+let eighteen = Uint32 18 in
+let decimals_at_least_6 = uint32_le six decimals in
+let decimals_no_more_than_18 = uint32_le decimals eighteen in
+andb decimals_at_least_6 decimals_no_more_than_18 in
+andb name_symbol_ok decimals_ok
+=>
 
 field implementation : ByStr20 = init_implementation
 field admin : ByStr20 = init_admin
-field balances : Map ByStr20 Uint128 = Emp ByStr20 Uint128
-field totalSupply : Uint128 = zero
+field balances : Map ByStr20 Uint128
+= let emp_map = Emp ByStr20 Uint128 in
+builtin put emp_map contract_owner init_supply
```

Comment:
Is this the one that we mentioned that is part of ZRC2 standard for initialising supply?

Reply:
yep